braindecode/datautil/windowers.py

```diff
@@ -134,7 +135,7 @@ class EventWindower(BaseWindower):
     ----------
     window_size_samples : int | None
         size of one window in samples
-    overlap_size_samples : int
+    stride_samples : int
         size of overlap for window in samples
     drop_last_samples : bool
         See BaseWindower
```

Comment:
@hubertjb @DavidSabbagh @sbbrandt  We now have a new Windower, see:
https://github.com/braindecode/braindecode/blob/1f01bbca020786af8cf3d5e14de941ae210952da/braindecode/datautil/windowers.py#L237-L301 this one computes supercrops outside of mne as it makes everything a bit easier for us our supercrop logic does not map well to mnes internal logic with chunk_duration.

Reply:
This looks good and is much clearer for me than the mne implementation before.

Reply:
@gemeinl Thanks! Please let me know if I understand correctly: you made a new class that reimplements what `EventWindower` was meant to do, but with the following differences: (1) it uses a new internal function (`_supercrop_starts`) instead of `mne.events_from_annotations` to identify events, (2) it uses 'crop'/'supercrop' language instead of 'windows' in the variable names and docstring, (3) it sets the default behaviour to `drop_samples=False` (instead of requiring the user to set this explicitly) and (4) it doesn't inherit from the `BaseWindower` class.

So, a couple of questions/points:

1. Before I forget, I think docstrings should start with a one-line short description on the same line as the first `"""`: see https://www.python.org/dev/peps/pep-0008/#documentation-strings
2. Do you see this new Windower as a replacement for `EventWindower` or is this meant as something more specific to the crop/supercrop case? If it is meant to replace `EventWindower`, why did you chose to go this route instead? Specifically, why did you chose to replace `mne.events_from_annotations`? Personally, I think the closer we stay to the MNE way of doing things, the simpler it will be to maintain the package.
3. If it is indeed meant to replace `EventWindower`, do you think we should keep the crop/supercrop-specific language? My worry is that since this class is central to the package, and these concepts are relatively niche, they might make the understanding of the library more difficult. As far as I understand, `supercrop` could just be replaced by `window` here as there is no subdivision into crops happening at this level.
4. I understand that you guys' use case is to extract a window that covers the last "missed" samples in each event. I still believe that this default behaviour can be problematic in some cases (as the opposite behaviour can be problematic in others). I liked our previous solution to this - what do you think?
5. I see there is commented code indicating that this class would also handle the use case for `FixedLengthWindower`. Are you planning on merging the two?

Reply:
I just saw the updated code - will repost updated comment below!